src/box/lua/schema.lua

```diff
@@ -812,18 +812,25 @@ box.schema.space.create = function(name, options)
             box.error(box.error.SPACE_EXISTS, name)
         end
     end
+    local _schema = box.space._schema
     local id = options.id
     if not id then
-        local _schema = box.space._schema
         local max_id = _schema:update({'max_id'}, {{'+', 2, 1}})
-        if max_id == nil then
-            id = _space.index.primary:max().id
+        id = _space.index.primary:max().id
+        if max_id == nil or id >= max_id.value then
             if id < box.schema.SYSTEM_ID_MAX then
                 id = box.schema.SYSTEM_ID_MAX
             end
-            max_id = _schema:insert{'max_id', id + 1}
+            max_id = _schema:replace{'max_id', id + 1}
         end
         id = max_id.value
+    else
+        local max_id_tup = _schema:get{'max_id'}
+        if max_id_tup ~= nil then
+            _schema:update({'max_id'}, {{'=', 2, math.max(max_id_tup[2], id)}})
```

Comment:
This worked before, now it doesn't:

```
tarantool> box.schema.space.create('test', {id = 1ULL})
---
- error: 'builtin/box/schema.lua:830: bad argument #2 to ''max'' (number expected,
    got cdata)'
...
```

Reply:
It's OK now, using this way of creation in the test.